input-forms.md

```diff
@@ -0,0 +1,82 @@
+# Input-forms Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**/api/configuration/input-forms**   
+
+Provide access to the configured input-forms. It returns the list of existent input-forms.
+Please note that starting from DSpace 7 each input-form will describe a single page of inputs, combining different pages togheter is done with the [submission-definition](submission-panels.md), aka the item-submission.xml configuration file 
```

Comment:
While I'm not against this concept of separate `input-forms.xml` files per "page", I wonder whether it will still be possible to specify a different input-form per *collection* (See `<form-map>` in [input-forms.xml](https://github.com/DSpace/DSpace/blob/master/dspace/config/input-forms.xml#L16))?  

I'm slightly worried that this won't be as easy as simply "splitting" apart existing `input-forms.xml` files, but also could affect the existing structure of `item-submission.xml`.  I'd like to better understand the proposal here, and how existing configs are proposed to be auto-updated. 

I'm also not entirely certain (myself) that auto-manipulating these old configurations into a new structure would be much *easier* than simply keeping them in the existing config structure and auto-generating `form-name`s like "traditional-1" (for form=traditional, page=1), "traditional-2", etc.

Reply:
@tomdesair  yes, we propose to remove the page level in the input-form dtd see https://github.com/DSpace/DSpace/pull/1852/files#diff-8f6c1e7efcbc9f92d1be349e0ed0270bL26
https://github.com/DSpace/DSpace/pull/1852/files#diff-0c10287e5ba0372e45a53a94118cf5d6L13

@tdonohue the input-form per collection functionality is preserved as also the item-submission.xml allows such kind of mapping. Assuming that you have in the current configuration:
- 2 processes defined in the item-submissions.xml 
- 2 input-forms defined in the input-forms.xml 
and you have collection1, collection2 using the process1, collection3 and collection4 using the process2. The collections with the odd ID using the inputform1 and the even inputform2 you will end with 4 (2x2) processes defined in the new item-submissions.xml
process1-inputform1 (collection1), process1-inputform2 (collection2), process1-inputform1 (collection3), process1-inputform2 (collection4)

We need to split the input-forms.xml by page as we need to have an ID to refer in the item-submissions.xml when we include a DescribeStep. This simplify the processing of the submission-definition that will end to be independent from the collection that use it and will make the whole configuration more readable. Please note that also the input-forms.xml file is likely to be simplified as I expect that some page definition will be shared between different existent input-forms (this is the case in many configuration that we have seen where the first pages have common metadata and the subsequent pages are more specialized).

Reply:
@abollini : I see your point.

I still think we'll have to be very careful with the upgrade process here. What you are defining is essentially a new configuration (even if some concepts are borrowed).  It sounds like, in the configuration you've described, the existing `input-forms.xml` would need to be split in two, but its collection specific configs would need to be _moved up_ into the existing `item-submissions.xml` config *AND* combined with other collection configs there in order to configure four separate processes/definitions in `item-submissions.xml`.

I do now see your point here in simplifying the configuration in this manner. But, I'm starting to wonder if we should rename these config files to make it crystal clear that it's a *new* configuration (even though it borrows from the old one).  For example, we could consider dropping the name `input-forms.xml` altogether, and prefix these new configs with `submission-form-[name].xml`.  The `item-submissions.xml` could keep the same name, since it's mostly just minimal updates.

Reply:
No problem to give a new name to the new format of input-forms. I propose to use submission-forms.xml. I'm uncertain about splitting each page in a separate file... do you see additional value on that? we will end with lot of files for large configuration and we will have also issue with the file name if the name assigned to the form is not file system friendly